laurat007/assignment1/Q1.java

```diff
@@ -32,102 +33,114 @@
 	
 	// number of pairs to be checked for being anagrams
 	static int N;
+        static Scanner sc;
 	
 	static boolean caseSensitiveAnagrams(String s1, String s2) {
-		System.out.println(s1 + " " + s2);
-		if(s1.length() != s2.length())
-			return false;
-		
-		// 58 = 122('z') - 65('A') + 1
-		int letters_occurrences[] = new int[58];
-		for(int i = 0 ; i < 58; i++)
-			letters_occurrences[i] = 0;
-		for(int i = 0; i < s1.length(); i++) {
-			letters_occurrences[122 - (int)s1.charAt(i)]++;
-		}
-		for(int i = 0; i < s1.length(); i++) {
-			letters_occurrences[122 - (int)s2.charAt(i)]--;
-		}
-		for(int i = 0 ; i < 57; i++)
-			if(letters_occurrences[i] != 0)
-				return false;
-		return true;
+            System.out.println(s1 + " " + s2);
+            if(s1.length() != s2.length())
+                return false;
+
+            // 58 = 122('z') - 65('A') + 1
+            int letters_occurrences[] = new int[58];
+            for(int i = 0 ; i < 58; i++)
+                letters_occurrences[i] = 0;
+            for(int i = 0; i < s1.length(); i++) {
+                letters_occurrences[122 - (int)s1.charAt(i)]++;
+            }
+            for(int i = 0; i < s1.length(); i++) {
+                letters_occurrences[122 - (int)s2.charAt(i)]--;
+            }
+            for(int i = 0 ; i < 57; i++)
```

Comment:
There should be 58 not 57 :) Can you define a variable 58 = 122('z') - 65('A') + 1 since you are reusing it many times? It is a common good practice and it would have prevented this error

Reply:
done defining a constant